pkg/domain/infra/abi/play.go

```diff
@@ -314,14 +314,36 @@ func (ic *ContainerEngine) PlayKube(ctx context.Context, body io.Reader, options
 		return nil, fmt.Errorf("YAML document does not contain any supported kube kind")
 	}
 
+	// If we started containers along with a service container, we are
+	// running inside a systemd unit and need to set the main PID.
 	if options.ServiceContainer && ranContainers {
-		message := fmt.Sprintf("MAINPID=%d\n%s", os.Getpid(), daemon.SdNotifyReady)
-		if err := notifyproxy.SendMessage("", message); err != nil {
-			return nil, err
-		}
+		switch len(notifyProxies) {
+		case 0: // Optimization for containers/podman/issues/17345
+			// No container needs sdnotify, so we can mark the
+			// service container as the main PID and return early.
+			data, err := serviceContainer.Inspect(false)
```

Comment:
Inspect is rather expensive since we look out for permanence I think it would be better to add an accessor for the ConmonPid.

Reply:
Nice idea!  With the afternoon meetings, I am running out of time but I can add an accessor in another PR.

We did the inspect before https://github.com/containers/podman/commit/4fa307f149234ed27a4e07da8065e1495a1962c2 so it's at least not a regression.

Reply:
sounds good